subsys/sal/sid_pal/src/sid_gpio_utils.c

```diff
@@ -78,6 +78,8 @@ uint32_t sid_gpio_utils_register_gpio(struct gpio_dt_spec gpio_from_dts)
 						      .pin = gpio_from_dts.pin },
 					    .configuration_cache = gpio_from_dts.dt_flags };
 
+	sid_gpio_utils_gpio_set_flags(ctx.next_free_slot,
```

Comment:
What are those flags for?

Reply:
This is to set the default flags of the gpio when it is registered for a first time. It should be redundant as we pull those flags from DTS, but I call it just to be sure that the state of new gpio is defined.  (previously it was hidden behind a static guard, so essentially we set this only on first GPIO registerd, but not for any other).

So to summarize, The change is not big, it only expands the behaviour to all GPIOs regardles of the order of registering, and as far as I understand this operation should not be even necessary.

We can remove it in next code refactoring session (if it won't break anything and tests pass) ;)